### ISSUE TITLE
[FromBody] parameters renamed to body

### ISSUE THREAD

**Issue by ThomasArdal** [domaindrivendev/Swashbuckle.AspNetCore#1767]
This is probably not a bug. But I noticed that Swashbuckle.AspNetCore renamed by `[FromBody]` parameters after upgrading from the ASP.NET version to Swashbuckle.AspNetCore. Like this:

```
public ActionResult<MyResponse> Post([FromBody] MyRequest request)
```

Would generate something like this in the old version:

```json
"parameters": [
{
  ...
  "name": "request",
  ...
],
```

But with Swashbuckle.AspNetCore, the name is set to `body`:

```json
"parameters": [
{
  ...
  "name": "body",
  ...
],
```

It's not a huge problem for me. But the generated client does have changes and if anyone would reference the request parameter by name, they would get a compile error.

Could you help with a short explanation of why this changed? And can I force Swashbuckle.AspNetCore to use the old naming?

**Comment by domaindrivendev** [comment]
As of v5.0.0, Swashbuckle made the jump to OpenAPI V3. As part of this change, it pulls in the OpenAPI.NET library and works exclusively with it's V3 object model. This means that when Swashbuckle generates Swagger it projects it into a V3 object model. If you have the `SerializeAsV2` flag enabled (which I'm assuming you do), then this V3 object model is mapped back to a V2 format at the time of serialization, using a feature provided by the [OpenAPI.NET library](https://github.com/microsoft/OpenAPI.NET).

In Swagger/OpenAPI V3, `body` parameters are no longer included in the `parameters` collection and are instead assigned to a new property called `requestBody`. This means there is no `name` for these parameters in the V3 spec and therefore there's no way to re-introduce a specific name when deserializing to V2 JSON. Instead, the serialize-as-v2 feature just defaults to a name of `body` for the body parameter.

**Comment by domaindrivendev** [comment]
Sounds like there might be something in the works over at [OpenAPI.NET] that would make this possible:

https://github.com/microsoft/OpenAPI.NET/issues/486

**Comment by ThomasArdal** [comment]
That is correct. I recently upgraded my API to Swashbuckle.AspNetCore v5 and OpenAPI v3 and my old C# client just worked (which is awesome). I then realized that I couldn't rebuild my client with AutoRest since that tool doesn't support v3 yet. So, to keep the new Swashbuckle.AspNetCore package I'm setting `SerializeAsV2` to `true` to generate something that AutoRest understands (until they support v3 or I switch to NSwag or something else).

Thank you so much for your explanation. Makes sense. I will have to live with the `body` name until microsoft/OpenAPI.NET#486 is hopefully implemented 👍
